common/include/Utilities/PString.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+#include "ghc/filesystem.h"
+#include <iostream>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <locale.h>
+#include <string>
+#include "Path.h"
+
+namespace fs = ghc::filesystem;
+
+class PString
+{
+private:
+	std::basic_string<char> string;
+public:
+	PString() = default;
+
+	// Non default constructors
+	PString(const char* string)
+	{
+		string = string;
+	}
+
+	explicit PString(const fs::path& path)
+	{
+		string = path.string();
+	}
+
+	PString(const wxString& str)
+	{
+		string = std::string(str.utf8_str());
+	}
+
+	#ifdef _WIN32
+	PString(const std::wstring& wString);
+	operator std::wstring();
+	#endif
+
+	#ifdef __cpp_lib_char8_t
+	PString(const std::u8string& str);
+	operator std::u8string();
+	#endif
+
+	// Copy Constructor
+	PString(const PString& rhs)
+	{
+		string = rhs.string;
+	}
+
+	PString(PString&& move)
+	{
+		string = move.string;
+	}
+
+	const bool operator==(const PString& rhs)
+	{
+		return string == rhs.string;
+	}
+
+	PString& operator=(PString&&);
+
+	// Copy Operator
+	const PString& operator=(const PString&);
+	
+	operator std::string() const;
+	operator wxString() const;
+	operator fs::path() const;
```

Comment:
These look quite scary, it might be better not to allow for implicit conversions to force the programmer to think twice about character conversions.

Reply:
Well most of them are using the inbuilt conversions or constructors to copy the string data over so it should be fine